src/main/java/nz/co/jammehcow/jenkinsdiscord/util/EmbedUtil.java

```diff
@@ -0,0 +1,352 @@
+                    listener,
+                    "matrix configuration",
+                    descriptionBuilder.toString().strip(),
+                    EmbedUtil.MAX_EMBED_FIELD_LENGTH
+            );
+
+            embedBuilder.addField(new EmbedField(false, "Configuration matrix", matrixDescription));
+        }
+
+        LinkedList<String> changesList = new LinkedList<>();
+        if (showChangeset) {
+            changesList = new LinkedList<>(EmbedUtil.buildChangesetList((RunWithSCM) build, scmWebUrlTemplate));
+        }
+
+        LinkedList<String> artifactsList = new LinkedList<>();
+        if (enableArtifactsList) {
+            artifactsList = new LinkedList<>(EmbedUtil.buildArtifactList(build, artifactsURL));
+        }
+
+
+        if (!changesList.isEmpty()) {
+            // shorten to at most 1024 characters (max field size)
+            while (EmbedUtil.getLengthForList(changesList) > EmbedUtil.MAX_EMBED_FIELD_LENGTH) {
+                changesList.removeLast();
+            }
+
+            StringBuilder changesDescription = new StringBuilder();
+            for (String changeEntry : changesList)
+                changesDescription.append(changeEntry)
+                                  .append('\n');
+
+            embedBuilder.addField(new EmbedField(false, "Changes", changesDescription.toString().strip()));
+        } else if (showChangeset) {
+            embedBuilder.addField(new EmbedField(false, "Changes", "*No changes.*"));
+        }
+
+        if (!artifactsList.isEmpty()) {
+            StringBuilder artifactsDescription = new StringBuilder();
+            for (String artifact : artifactsList)
+                artifactsDescription.append(artifact)
+                                    .append('\n');
+
+            if (artifactsDescription.length() > EmbedUtil.MAX_EMBED_FIELD_LENGTH)
+                embedBuilder.addField(new EmbedField(false, "Artifacts", artifactsURL));
+            else
+                embedBuilder.addField(new EmbedField(false, "Artifacts", artifactsDescription.toString().strip()));
+        } else if (enableArtifactsList) {
+            embedBuilder.addField(new EmbedField(false, "Artifacts", "*No artifacts saved.*"));
+        }
+
+        embedBuilder.setColor(statusColor.getCode());
+        embedBuilder.setImageUrl(image);
+        embedBuilder.setThumbnailUrl(thumbnail);
+
+        if (footer != null) {
+            EmbedFooter embedFooter = new EmbedFooter(
+                    EmbedUtil.truncateToLimit(listener, "footer", footer, EmbedUtil.MAX_EMBED_FOOTER_LENGTH),
+                    null
+            );
+
+            embedBuilder.setFooter(embedFooter);
+        }
+
+        String username = EmbedUtil.withFallback(customUsername, "Jenkins");
+        String avatar = EmbedUtil.withFallback(customAvatarUrl, "https://get.jenkins.io/art/jenkins-logo/1024x1024/headshot.png");
+        embedBuilder.setAuthor(new EmbedAuthor(username, avatar, null));
```

Comment:
Username and avatar should also be set on the webhook message to match previous functionality.